CI on FreeBSD is failing due to regression in `libz-sys`

Observed in https://github.com/rust-lang/rustup/pull/4758#issuecomment-4083805979:

> I think in principle this should work but I have encountered another regression in `libz-sys` (maybe caused by https://github.com/rust-lang/rustup/pull/4753)... so now we have 3 different issues for FreeBSD now!

```smalltalk
cargo:warning=ToolExecError: command did not execute successfully (status code exit status: 1): LC_ALL="C" "cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "--target=x86_64-unknown-freebsd" "-I" "src/zlib" "-w" "-fvisibility=hidden" "-DSTDC" "-D_LARGEFILE64_SOURCE" "-D_POSIX_SOURCE" "-o" "/home/runner/work/rustup/rustup/target/x86_64-unknown-freebsd/release/build/libz-sys-046af476f6182d2e/out/lib/f0389296f42960e9-gzread.o" "-c" "src/zlib/gzread.c"cargo:warning=src/zlib/gzwrite.c:82:49: error: use of undeclared identifier 'EWOULDBLOCK'
    cargo:warning=                if (errno == EAGAIN || errno == EWOULDBLOCK)
    cargo:warning=                                                ^
    cargo:warning=src/zlib/gzwrite.c:117:53: error: use of undeclared identifier 'EWOULDBLOCK'
    cargo:warning=                    if (errno == EAGAIN || errno == EWOULDBLOCK)
    cargo:warning=                                                    ^
    cargo:warning=2 errors generated.
  
    exit status: 1
    cargo:warning=ToolExecError: command did not execute successfully (status code exit status: 1): LC_ALL="C" "cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "--target=x86_64-unknown-freebsd" "-I" "src/zlib" "-w" "-fvisibility=hidden" "-DSTDC" "-D_LARGEFILE64_SOURCE" "-D_POSIX_SOURCE" "-o" "/home/runner/work/rustup/rustup/target/x86_64-unknown-freebsd/release/build/libz-sys-046af476f6182d2e/out/lib/f0389296f42960e9-gzwrite.o" "-c" "src/zlib/gzwrite.c"
```
_https://github.com/rust-lang/rustup/actions/runs/23254426636/job/67605187428?pr=4758_

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -1361,7 +1361,7 @@ jobs:
         usesh: true
         copyback: false
         prepare: |
-          pkg install -y git gmake bash sudo \
+          pkg install -y git gmake bash sudo ca_root_nss \
             `# The following packages are required by 'aws-lc-rs':` \
             cmake-core llvm-devel-lite rust-bindgen-cli
         run: |
@@ -1393,7 +1393,7 @@ jobs:
         usesh: true
         copyback: false
         prepare: |
-          pkg install -y git gmake bash sudo \
+          pkg install -y git gmake bash sudo ca_root_nss \
             `# The following packages are required by 'aws-lc-rs':` \
             cmake-core llvm-devel-lite rust-bindgen-cli
         run: |

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -137,6 +137,12 @@ features = [
 ]
 version = "0.61"
 
+# HACK: Temporarily pinned due to `zlib` regression, to be removed when
+# <https://github.com/madler/zlib/issues/1168> is resolved.
+# See: <https://github.com/rust-lang/rustup/issues/4760>
+[target.'cfg(target_os = "freebsd")'.dependencies]
+libz-sys = "=1.1.24"
+
 [dev-dependencies]
 enum-map = "2.5.0"
 http-body-util = "0.1.0"

--- a/ci/actions-templates/freebsd-builds-template.yaml
+++ b/ci/actions-templates/freebsd-builds-template.yaml
@@ -22,7 +22,7 @@ jobs: # skip-main skip-stable
         usesh: true
         copyback: false
         prepare: |
-          pkg install -y git gmake bash sudo \
+          pkg install -y git gmake bash sudo ca_root_nss \
             `# The following packages are required by 'aws-lc-rs':` \
             cmake-core llvm-devel-lite rust-bindgen-cli
         run: |

